quincy: admin/doc-requirements: bump sphinx to 4.4.0

ljflores opened the pull request:
backport tracker: https://tracker.ceph.com/issues/55051

---

backport of https://github.com/ceph/ceph/pull/45266
parent tracker: https://tracker.ceph.com/issues/55050

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

ljflores commented:
jenkins test make check

ljflores commented:
Ugh, still some dependency issues. Looking into what's going on.

ljflores commented:
jenkins test api

rzarzynski commented on doc/conf.py:
```diff
@@ -167,7 +167,8 @@ def is_release_eol(codename):
     'MACRO_EXPANSION': 'YES',
-    'PREDEFINED': 'CEPH_RADOS_API= '
+    'PREDEFINED': 'CEPH_RADOS_API= ',
+    'WARN_IF_UNDOCUMENTED': 'NO',
```
:+1: for the ending comma.

rzarzynski commented on src/python-common/ceph/deployment/service_spec.py:
```diff
@@ -314,28 +314,34 @@ def from_string(cls, arg):
         PlacementSpec(label='mon')
 
-        Labels als support a count:
+        Labels also support a count:
```
ACK.

rzarzynski reviewed (approved):
Although I'm not a sphinx expert, I'm not seeing any obvious issues here.

tchaikov commented on src/pybind/mgr/dashboard/constraints.txt:
```diff
@@ -3,5 +3,5 @@ more-itertools==4.1.0
 PyJWT==2.0.1
 bcrypt==3.1.4
 python3-saml==1.4.1
-requests==2.25.1
```
might want to introduce this change to master first, and then backport it to this branch.

djgalloway replied:
And, actually, wait an hour or so because you might be able to cherry-pick from https://github.com/ceph/ceph/pull/45647

ljflores replied:
Will do, removed that commit and marked this PR with DNM.

ljflores commented:
I think @djgalloway is already taking care of this in the proper workflow (#45646). I'll close this PR if his fix resolves the problem.

djgalloway commented:
I don't think my PR has enough to fix the errors/warnings.  I'll defer to you.

djgalloway commented:
Yeah, this looks good aside from https://github.com/ceph/ceph/pull/45627#discussion_r835434857.  Let's remove the `requests` change for now and we'll backport whatever happens in https://github.com/ceph/ceph/pull/45647 separately.

ljflores commented:
> Yeah, this looks good aside from [#45627 (comment)](https://github.com/ceph/ceph/pull/45627#discussion_r835434857). Let's remove the `requests` change for now and we'll backport whatever happens in #45647 separately.

Sounds good, I'll put make a note to merge this one after #45647.

ljflores commented:
> Oh, I meant we can merge this and address the other dependency separately.  All of these changes are still needed in quincy.

Oh I see.

djgalloway commented:
This has failed the dashboard test twice in a row the same way which makes me hesitant to merge.

```
The error we received was:

Error: [0mEyes-Cypress detected diffs or errors during execution of visual tests.[0m[0m See details at: [38;5;86mhttps://eyes.applitools.com/app/batches/00000251754060886964/00000251754060886763?accountId=hCTedwnSTUWSCZlXPqua0g~~[0m[0m
       [32mPassed - 7 tests[0m
         [32m✓[0m [0mDashboard Component [1920x1080][0m
         [32m✓[0m [0mDashboard Component [800x600][0m
         [32m✓[0m [0mDashboard Component [800x600][0m
         [32m✓[0m [0mLogin Component Check [1920x1080][0m
         [32m✓[0m [0mLogin Component Check [1920x1080][0m
         [32m✓[0m [0mLogin Component Check [800x600][0m
         [32m✓[0m [0mLogin Component Check [800x600][0m
       [38;5;214mDiffs detected - 1 tests[0m
         [38;5;214m⚠[0m [0mDashboard Component [1920x1080][0m

       [0mSee details at:[0m [38;5;86mhttps://eyes.applitools.com/app/batches/00000251754060886964/00000251754060886763?accountId=hCTedwnSTUWSCZlXPqua0g~~[0m

    at /home/jenkins-build/build/workspace/ceph-dashboard-pull-requests/src/pybind/mgr/dashboard/frontend/node_modules/@applitools/eyes-cypress/src/plugin/waitForBatch.js:26:13
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at Object.handler (/home/jenkins-build/build/workspace/ceph-dashboard-pull-requests/src/pybind/mgr/dashboard/frontend/node_modules/@applitools/eyes-cypress/src/plugin/hooks.js:35:7)

2022-03-25T20:25:48.624+0000 7febabd6f340 -1 WARNING: all dangerous and experimental features are enabled.
2022-03-25T20:25:48.624+0000 7febabd6f340 -1 WARNING: all dangerous and experimental features are enabled.
2022-03-25T20:25:48.660+0000 7f1e968d4340 -1 WARNING: all dangerous and experimental features are enabled.
2022-03-25T20:25:48.660+0000 7f1e968d4340 -1 WARNING: all dangerous and experimental features are enabled.

WARNING:  ceph-osd ceph-mds still alive after 1 seconds

WARNING:  ceph-osd ceph-mds still alive after 2 seconds

WARNING:  ceph-osd ceph-mds still alive after 4 seconds

WARNING:  ceph-mds still alive after 7 seconds

Build step 'Execute shell' marked build as failure
```

ljflores commented:
@djgalloway the same thing is failing in #45647, so I think it is a separate issue. Also, the current commits do not make any changes to dashboard dependencies-- the only reason that test was triggered is because I introduced the extra commit that bumped the version of "requests", which is now removed.

Would still be good to have dashboard approval on this though. @epuertat @nizamial09

epuertat commented:
jenkins test dashboard
